lib/std/heap/general_purpose_allocator.zig

```diff
@@ -157,6 +162,9 @@ pub const Config = struct {
 
 pub const Check = enum { ok, leak };
 
+var used_small_bucket_count_cache = std.atomic.Value(usize).init(0);
+var largest_used_bucket_object_size_cache = std.atomic.Value(usize).init(0);
```

Comment:
you can't use global state like this, this API is not a singleton. this is totally broken.

not to worry, however, I'm going to fix it for you and then merge, so just sit tight :+1: